src/luascript.cpp

```diff
@@ -11654,6 +11655,24 @@ int LuaScriptInterface::luaItemTypeGetDecayId(lua_State* L)
 	return 1;
 }
 
+int LuaScriptInterface::luaItemTypeGetDecayTime(lua_State* L)
+{
+	// itemType:getDecayTime()
+	const ItemType* itemType = getUserdata<const ItemType>(L, 1);
+	if (itemType) {
+		uint32_t decayTime = itemType->decayTime;
+		if (decayTime == 0 && itemType->showDuration) {
+			const ItemType& equipItemType = Item::items[itemType->transformEquipTo];
```

Comment:
this wont work for items that are not transformed on equip to anything else right?

Reply:
16:08 > print(ItemType(2050):getDecayTime()) -> 0 
16:08 > print(ItemType(2051):getDecayTime()) -> 600
16:09 > print(ItemType(2052):getDecayTime()) -> 0
16:09 > print(ItemType(2053):getDecayTime()) -> 300

what other flag could be used to return correct duration for just bought torch?

Reply:
eventually we can do two different functions, like
- `:getDecayBaseTime()` would work as current getDecayTime()
- `:getDecayTime()` would check lua libs as well

otherwise it's probably not the greatest idea to hardcode a variable from some random lua file